File: app/src/main/java/sfedu/physics/dmitriy/githubapiresttestproject/mvp/users/UsersView.java
package sfedu.physics.dmitriy.githubapiresttestproject.mvp.users;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

import sfedu.physics.dmitriy.githubapiresttestproject.user_model.UserResponse;

@StateStrategyType(AddToEndStrategy.class)
public interface UsersView extends MvpView {
    void onStartLoading();

    void loadUsersByQuery(UserResponse userResponse);
    void onErrorLoadingUsers(Throwable throwable);
    void onLoadingFinished();

    void loadMoreUsers(UserResponse userResponse);
}

File: app/src/main/java/sfedu/physics/dmitriy/githubapiresttestproject/mvp/users/UsersView.kt
package sfedu.physics.dmitriy.githubapiresttestproject.mvp.users

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

import sfedu.physics.dmitriy.githubapiresttestproject.json_model.user_model.UserResponse

@StateStrategyType(AddToEndStrategy::class)
interface UsersView : MvpView {
    fun onStartLoading()

    fun loadUsersByQuery(userResponse: UserResponse)
    fun onErrorLoadingUsers(throwable: Throwable)
    fun onLoadingFinished()

    fun loadMoreUsers(userResponse: UserResponse)
}
